core/controllers/library.py

```diff
@@ -215,15 +214,24 @@ def get(self):
             (ord(char), None) for char in string.punctuation)
         query_string = query_string.translate(remove_punctuation_map)
 
-        if self.request.get('category'):
-            query_string += ' category=%s' % self.request.get('category')
-        if self.request.get('language_code'):
-            query_string += ' language_code=%s' % self.request.get(
-                'language_code')
+        # If there is a category parameter, it will be in the following form:
+        #     category=("Algebra" OR "Math")
+        category_string = self.request.get('category', '')
+        categories = (
+            category_string[2:-2].split('" OR "') if category_string else [])
```

Comment:
Should we use re.match('\"\s*\"', category_string) and create groups? (Just to avoid random number 2, `"OR"` if-else, not sure!)

Reply:
I don't think you can do this with a variable number of groups. But I have added some explanation inline, hopefully that helps.

Reply:
I meant something like this:
![image](https://user-images.githubusercontent.com/16653571/103180645-9b82ee00-48bd-11eb-80af-f03064503df7.png)

I think this will be simpler if above (line 240) we are going to have a regex check to ensure that the entire string only contains OR (not sure!).

Reply:
I don't think this is simpler. I think splitting is the right thing to do here because this is more about parsing the string (a splitter is a simple parser) than just matching it to a pattern.